src/app/routes/cpsAsset/utils/getAdditionalPageData.js

```diff
@@ -73,18 +126,48 @@ const pageTypeUrls = async (
   }
 };
 
-const validateResponse = ({ status, json }, name) => {
+const validateResponse = ({ status, json }, name, engine) => {
   if (status === 200 && !isEmpty(json)) {
-    return { [name]: json };
+    // 004_brasil_recommendations_experiment
+    return engine ? { [name]: { [engine]: json } } : { [name]: json };
   }
 
   return null;
 };
 
-const fetchUrl = ({ name, path, ...loggerArgs }) =>
-  fetchPageData({ path, timeout: SECONDARY_DATA_TIMEOUT, ...loggerArgs })
-    .then(response => validateResponse(response, name))
+// 004_brasil_recommendations_experiment
+const sortAdditionalPageData = results => {
```

Comment:
I'm not exactly sure what this logic is doing? Is there an explict test for it? Just a bit wary it could break features reliant on additional data so be good to understand what it is doing. I could even warrant seperating into its own module with tests?

Reply:
looks much better now, nice work!